src/video/win32_v.cpp

```diff
@@ -1316,3 +1316,45 @@ Dimension VideoDriver_Win32::GetScreenSize() const
 {
 	return { static_cast<uint>(GetSystemMetrics(SM_CXSCREEN)), static_cast<uint>(GetSystemMetrics(SM_CYSCREEN)) };
 }
+
+ZoomLevel VideoDriver_Win32::GetSuggestedUIZoom()
+{
+	typedef UINT (WINAPI *PFNGETDPIFORWINDOW)(HWND hwnd);
+	typedef UINT (WINAPI *PFNGETDPIFORSYSTEM)(VOID);
+	typedef HRESULT (WINAPI *PFNGETDPIFORMONITOR)(HMONITOR hMonitor, int dpiType, UINT *dpiX, UINT *dpiY);
```

Comment:
Is this normal for OpenTTD, to have these kind of `typedef`s inside a function?

Reply:
I think it's reasonable to do for this case at least. When you're feature-detecting for functions that may only exist in a higher API level than you're statically linking for, it makes sense to make the scope as narrow as possible.